### PR TITLE
Allow adding new files to an empty repo

### PR DESCRIPTION
**Pull request by wxiaoguang**
image

### CODE REVIEW

File: modules/git/repo.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func InitRepository(ctx context.Context, repoPath string, bare bool) error {
 // IsEmpty Check if repository is empty.
 func (repo *Repository) IsEmpty() (bool, error) {
 	var errbuf, output strings.Builder
-	if err := NewCommand(repo.Ctx, "show-ref", "--head", "^HEAD$").
+	if err := NewCommand(repo.Ctx).AddOptionFormat("--git-dir=%s", repo.Path).AddArguments("show-ref", "--head", "^HEAD$").
 		Run(&RunOpts{
 			Dir:    repo.Path,
```
Comment (lunny) [inline modules/git/repo.go:85]: So the `Dir` can be removed here?

Comment (wxiaoguang) [reply modules/git/repo.go:85]: I'd like to keep it, leave it to future refactoring.


Actually, I think we need both `Dir` and `--git-dir` in the futre, `Dir` purpose is to change the current working directory.

File: routers/web/repo/view.go [hunk 2]
```diff
@@ -927,12 +929,7 @@ func renderCode(ctx *context.Context) {
 		return
 	}
 
-	if !ctx.Repo.Repository.IsEmpty {
-		checkCitationFile(ctx, entry)
-		if ctx.Written() {
-			return
-		}
```
Comment (wolfogre) [inline routers/web/repo/view.go:934]: Why was this removed?

Comment (wxiaoguang) [reply routers/web/repo/view.go:934]: My mistake. Fixed in 9c503735a91e5f8d0bdf8b52a7a14978c5170b95

**Review by lunny — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]
